packages/manager/modules/pci/src/projects/project/storages/containers/containers.controller.js

```diff
@@ -1,12 +1,47 @@
+import set from 'lodash/set';
+
 export default class PciStoragesContainersController {
   /* @ngInject */
-  constructor($translate, CucCloudMessage) {
+  constructor($translate, $http, CucCloudMessage) {
     this.$translate = $translate;
+    this.$http = $http;
     this.CucCloudMessage = CucCloudMessage;
   }
 
   $onInit() {
     this.loadMessages();
+    this.publicToggleLoading = false;
+  }
+
+  onPublicToggle(container) {
+    set(container, 'publicToggleLoading', true);
```

Comment:
We shouldn't update the parameter value. Please create a copy or handle it differently